app/controllers/projects_controller.rb

```diff
@@ -38,7 +39,29 @@ def show
   end
 
   # GET /projects/1/edit
-  def edit; end
+  def edit
+    return unless Flipper.enabled?(:suggest_tags)
+
+    @suggested_tags = []
+    # generating tags from the circuit elements used
+    circuit_data = JSON.parse(@project.project_datum.data)
```

Comment:
Extract this logic to ruby class and place in app/services. Also include tests for that